controllers/helmrepository_controller_test.go

```diff
@@ -523,6 +525,25 @@ func TestHelmRepositoryReconciler_reconcileSource(t *testing.T) {
 			},
 			want: sreconcile.ResultSuccess,
 		},
+		{
+			name:     "memory cache will be set when reconciler comes with cache and it is not configured with chartRepo",
+			protocol: "http",
+			cache:    &cache.Cache{},
+			want:     sreconcile.ResultSuccess,
+			assertConditions: []metav1.Condition{
+				*conditions.TrueCondition(sourcev1.ArtifactOutdatedCondition, "NewRevision", "new index revision"),
+				*conditions.TrueCondition(meta.ReconcilingCondition, "NewRevision", "new index revision"),
+			},
+			beforeFunc: func(t *WithT, memCache *cache.Cache, obj *sourcev1.HelmRepository, checksum string) {
+				memCache = cache.New(3, 60)
+			},
+			afterFunc: func(t *WithT, obj *sourcev1.HelmRepository, artifact sourcev1.Artifact, chartRepo repository.ChartRepository) {
+				t.Expect(chartRepo.IndexCache).ToNot(BeNil())
+				t.Expect(chartRepo.IndexKey).To(Equal(chartRepo.CachePath))
```

Comment:
can you also verify that the `IndexKey` is a valid path and not ""?

Reply:
added an assert [here](https://github.com/fluxcd/source-controller/pull/698/commits/08ef783ab587f57d78355a8b385f37cee968f3fe#diff-80d1918f4177a4ddcc50bfb633ae3c300f09da5dad2c4c18fea182a37a406997R542) to verify the path exists (which rules out the "" path.)